sklearn/base.py

```diff
@@ -297,15 +298,8 @@ def __setstate__(self, state):
             pickle_version = state.pop("_sklearn_version", "pre-0.18")
             if pickle_version != __version__:
                 warnings.warn(
-                    "Trying to unpickle estimator {0} from version {1} when "
-                    "using version {2}. This might lead to breaking code or "
-                    "invalid results. Use at your own risk. "
-                    "For more info please refer to:\n"
-                    "https://scikit-learn.org/stable/model_persistence.html"
-                    "#security-maintainability-limitations".format(
-                        self.__class__.__name__, pickle_version, __version__
-                    ),
-                    UserWarning,
+                    f"{self.__class__.__name__}|{__version__}|{pickle_version}",
+                    InconsistentVersionWarning,
```

Comment:
why not doing a `warnings.warn(InconsistentVersionWarning(name, version, pickle_version))` instead, and avoid parsing the message string?

Reply:
Because... I did not know that was possible 😂. The string parsing was a hack. Thank you for the tip!